model-single-origin-samples/models/model6.5_negbinom_matrix_correlation_features_oos_optim_otherbucket_samplevariance.stan

```diff
@@ -0,0 +1,149 @@
+        sample2_y[sample2[n], gene2[n]] = y2[n];
+    }
+    nu = 1;
+}
+parameters {
+    cholesky_factor_corr[C] Omega_L;
+    vector<lower=0>[C] Omega_sigma;
+    //corr_matrix[C] Omega;        // degree of correlation among loading factors for each cell type
+    //vector<lower=0>[C] tau;      // scale for each cell type - multiplied (on diagonal) with Omega
+    
+    matrix<lower=0>[G, C] theta; // loading factors for each gene, for each cell type
+    vector[C] theta_mu;          // mean expression level for each cell type
+    vector[M] theta_coefs_raw;
+    vector[M] theta_coefs_per_gene[G];
+    
+    vector[G] log_gene_base;     // constant intercept expression level for each gene, irrespective of cell type
+    // vector<lower=0>[G] gene_phi; // overdispersion parameter per transcript (for now)
+    simplex[C] sample2_x[S2];     // inferred sample2 compositions (simplex type enforces sum-to-one)
+
+    vector<lower=0, upper=1>[S2] unknown_prop; // proportion of each test sample that is of unknown cell type
+    vector[G] other_log_contribution_per_gene[S2]; // for each test sample, per-transcript contribution of unknown cell type
+
+    // hierarchical dispersion
+    real log_global_phi_scale;      // single global-scale value that doesn't vary by sample, gene, or cell type.
+    simplex[G] log_gene_phi;        // constant intercept overdispersion parameter per transcript. simplex solves identifiability problem
+    vector[S] log_sample_scale;     // sample-level variance
+    vector[S2] log_sample2_scale;   // sample2-level variance
+    matrix[G, C] celltype_scale;    // cell type-level variance (varies by transcript)
+
+}
+transformed parameters {
+    vector[M] theta_coefs[G];
+    vector[G] corrected_phis[S];
+    vector[G] corrected_phis2[S2];
+    vector[G] ones;
+    
+    for (g in 1:G) 
+        theta_coefs[g] = theta_coefs_raw + theta_coefs_per_gene[g];
+
+    for (s in 1:S)
+        corrected_phis[s] = log_global_phi_scale + log_gene_phi + log_sample_scale[s] + log(celltype_scale * sample_x[s]);
+    corrected_phis = exp(corrected_phis);
+
+    for (s in 1:S2)
+        corrected_phis2[s] = log_global_phi_scale + log_gene_phi + log_sample2_scale[s] + log(celltype_scale * sample_x[s]);
+    corrected_phis2 = exp(corrected_phis2);
+
+    for(g in 1:G)
+        ones[g] = 1; // initialize a vector of all ones (for dirichlet prior)
+}
+model {
+    // estimate theta - gene-level expression per cell type, as a function of cell-surface expression proteins
+    theta_mu ~ normal(0, 1);
+    theta_coefs_raw ~ normal(0, 1);
+    Omega_sigma ~ gamma(0.1, 0.1);
+    Omega_L ~ lkj_corr_cholesky(nu);
+    for (g in 1:G) {
+        vector[C] theta_tmp; // temporary predictor for cell-gene-specific expression level
+        theta_coefs_per_gene[g] ~ normal(0, 1);
+        theta_tmp = theta_mu + cell_features*theta_coefs[g];
+        theta[g] ~ multi_normal_cholesky(theta_tmp, diag_pre_multiply(Omega_sigma, Omega_L));
```

Comment:
should prob do `diag_pre_multiply` part outside of the `for (g in 1:G)` loop. computing this for every gene could get expensive.